lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -41,14 +41,19 @@ def run(delay_sync: 3.seconds, reminder_interval: 30.seconds)
     def report_what_just_happened(resources, watch_time)
       new_successes, new_failures = resources.partition(&:deploy_succeeded?)
       new_failures.each do |resource|
+        tags = %W(namespace:#{@namespace} sha:#{ENV['REVISION']} resource:#{resource.id})
         if resource.deploy_failed?
+          ::StatsD.increment('failure', 1, tags: tags)
```

Comment:
I think we should make it more obvious which metrics deal with a whole deploy task and which are about subresources, while keeping what we emit about each somewhat parallel. Maybe `KubernetesDeploy.deploy.status` vs `KubernetesDeploy.deploy.resource.status` and have success/failure/timeout as tags? And then similarly `KubernetesDeploy.deploy.duration` etc. (The "deploy" part is because this gem can also do restarts and task runs) Wdyt?